### PR TITLE
Feature/coovachilli

### PR DESCRIPTION
**Pull request by dwlfrth**
# Description

Add CoovaChilli support
# Impacts
- External captive portal detection
- Switches modules
# Delete branch after merge

YES
# NEWS file entries
## New Features
- Add support for CoovaChilli capable hardware

### CODE REVIEW

File: docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    See PacketFence_CoovaChilli_Quick_Integration_Guide-docinfo.xml for
+    authors, copyright and license information.
+
+////
+
+include::includes/global-attributes.asciidoc[]
+
+About this Guide
+----------------
+This guide has been created in order to help setting up a consumer grade access-point running CoovaChilli integration with PacketFence to use UAM capabilities along with PacketFence feature set. It can also help sales engineers, product managers, or network specialists demonstrate the PacketFence capabilities on-site with an existing or potential customer, or, provide guidelines to setup a proof of concept for a potential PacketFence deployment using CoovaChilli capable products.
```
Comment (extrafu) [inline docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc]: access point

File: docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+The instructions are based on version {release_version} of the PacketFence Zero Effort NAC (ZEN) image.
+
+Assumptions
+-----------
+* You have a CoovaChilli capable access-point running OpenWRT, on which CoovaChilli is installed (CoovaChilli installation is not covered in this guide);
```
Comment (extrafu) [inline docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc]: access point => everywhere else required

Comment (extrafu) [inline docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc]: Minimal versions required for OpenWRT and CoovaChilli?

File: docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+Having the ZEN image on hand, load it onto the virtual host and fire it up. You should be able to access it using 'https://IP_ADDRESS:1443' which will bring the PacketFence configurator
+
+Step 1: Basic PacketFence and network installation / configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+This step again, is only required in the case you do not already have a working (and configured) PacketFence installation. If it's the case, you might want to go ahead and proceed with step 2.
+
+Network interconnectivity
+^^^^^^^^^^^^^^^^^^^^^^^^^
+For the scope of this guide, the following network setup will be configured. Please note that a more complex setup can easily be used but will not be covered in this guide.
```
Comment (extrafu) [inline docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc]: more complex configuration

File: docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc (hunk 2 above)
Comment (dwlfrth) [inline docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc]: None that I'm aware of...

File: docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+Step 6: Services & Confirmation
++++++++++++++++++++++++++++++++
+The last but not the least. Here, we start the PacketFence server according to the configurations made in the previous steps. If everything goes as expected, you'll be prompted by a window inviting you to continue to the web administration interface.
+
+You'll be able to login to the PacketFence web administration interface with the credentials created in Step 5.
+
+Services status will help you monitor if everything goes as expected. If not, you'll see which service is in trouble and the log output will help you determine the problem that occurs.
+
+Step 2: Access-point and CoovaChilli configuration
```
Comment (extrafu) [inline docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc]: Access Point and CoovaChilli Configuration

Comment (extrafu) [inline docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc]: Can be a bit confusing, we were at Step 5, Step 6 just before...

File: docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Now that we have a functionnal PacketFence installation, we will go ahead and start by configuring the access point and CoovaChilli running on it. We will then use some of the configuration parameters to finish PacketFence integration in step 3.
+
+This guide assume that CoovaChilli is installed on the access point. If it's not, we suggest you Google around and make sure to find the appropriate way to do so since there's just to much vendor and network equipment that supports CoovaChilli to document this step here.
+
+The guide also assume that you have an SSID configured on the access point.
```
Comment (extrafu) [inline docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc]: assumes

File: docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+
+You should also make sure to have a default route properly configured on the access point (so that it can access the Internet) and that DNS resolution is working.
+
+Also note that manipulation on the OpenWRT access point are done using SSH shell access.
```
Comment (extrafu) [inline docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc]: manipulation -> changes

File: docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+This guide has been created in order to help setting up a consumer grade access point running CoovaChilli integration with PacketFence to use UAM capabilities along with PacketFence feature set. It can also help sales engineers, product managers, or network specialists demonstrate the PacketFence capabilities on-site with an existing or potential customer, or, provide guidelines to setup a proof of concept for a potential PacketFence deployment using CoovaChilli capable products.
+
+The instructions are based on version {release_version} of the PacketFence Zero Effort NAC (ZEN) image.
+
+Assumptions
+-----------
+* You have a CoovaChilli capable access point running OpenWRT, on which CoovaChilli is installed (CoovaChilli installation is not covered in this guide);
+* You have a virtual "host" on which it is possible to load the latest PacketFence ZEN image;
+* Network interconnection between the virtual "host" running the latest PacketFence ZEN image, the CoovaChilli capable access point, and Internet is functional;
+* A PacketFence webauth enforcement setup will be deployed;
```
Comment (extrafu) [inline docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc]: webauth -> WebAuth   everywhere?

File: docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc (hunk 4 above)
Comment (dwlfrth) [inline docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc]: With asciidoc indentation it should be better...

File: docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+Also note that changes on the OpenWRT access point are done using SSH shell access.
+
+Please note that any interface name reference might be different from one hardware to another.
+
+CoovaChilli requires some slight modifications to the way networking is working on the OpenWRT access point.
+
+ * The LAN and WLAN interface are bridged
+ * There is a DHCP server running on the LAN which can potentially wreak havoc
+
+The followings needs to be done:
```
Comment (extrafu) [inline docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc]: The following steps need to be done:

File: docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+
+ * Remove bridge between 'eth0' and 'wlan0' and assign device's IP to 'eth0'
+ * Disable 'dnsmasq' service which act as DHCP and DNS servers
+ * Configure chilli
```
Comment (extrafu) [inline docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc]: 'chilli'

File: lib/pf/Switch/CoovaChilli.pm [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+
+pf::Switch::CoovaChilli
+
+=head1 SYNOPSIS
+
+Implement object oriented module to handle CoovaChilli network equipment
+
+=head1 STATUS
+
+Tested and developped on a Ubiquiti NanoStation M2 running OpenWRT CHAOS CALMER (15.05.1,r48532)
```
Comment (extrafu) [inline lib/pf/Switch/CoovaChilli.pm]: developed

File: docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc [hunk 11]
```diff
@@ -0,0 +1,209 @@
...
+
+You'll be able to login to the PacketFence web administration interface with the credentials created in Step 5.
+
+Services status will help you monitor if everything goes as expected. If not, you'll see which service is in trouble and the log output will help you determine the problem that occurs.
+
+Step 2: Access Point and CoovaChilli Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Now that we have a functionnal PacketFence installation, we will go ahead and start by configuring the access point and CoovaChilli running on it. We will then use some of the configuration parameters to finish PacketFence integration in step 3.
+
+This guide assume that CoovaChilli is installed on the access point. If it's not, we suggest you Google around and make sure to find the appropriate way to do so since there's just to much vendor and network equipment that supports CoovaChilli to document this step here.
```
Comment (extrafu) [inline docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc]: ... we suggest you search relevant information on the Internet to install CoovaChilli as there are too many network equipment vendors that support CoovaChilli to accurately document this step here.

File: docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc [hunk 12]
```diff
@@ -0,0 +1,209 @@
...
+
+The guide also assumes that you have an SSID configured on the access point.
+
+You should also make sure to have a default route properly configured on the access point (so that it can access the Internet) and that DNS resolution is working.
+
+Also note that changes on the OpenWRT access point are done using SSH shell access.
+
+Please note that any interface name reference might be different from one hardware to another.
```
Comment (extrafu) [inline docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc]: from one equipment vendor to an other.

File: docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc [hunk 13]
```diff
@@ -0,0 +1,209 @@
...
+
+CoovaChilli requires some slight modifications to the way networking is working on the OpenWRT access point.
+
+ * The LAN and WLAN interface are bridged
+ * There is a DHCP server running on the LAN which can potentially wreak havoc
```
Comment (extrafu) [inline docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc]: can potentially cause major problems on production networks

File: docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc [hunk 14]
```diff
@@ -0,0 +1,209 @@
...
+
+Disable DHCP/DNS server
++++++++++++++++++++++++
+Execute the following to make sure 'dnsmasq' service is not started at boot
+
+    /etc/init.d/dnsmasq disable
+
+Configure chilli
+++++++++++++++++
+chilli configuration might differ from one hardware to another. Just make sure to follow theses configuration guidelines and you should be all-set.
```
Comment (extrafu) [inline docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc]: from one equipment vendor to an other one.

Comment (extrafu) [inline docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc]: these

File: docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc [hunk 15]
```diff
@@ -0,0 +1,209 @@
...
+    HS_RADSECRET    The RADIUS secret that will be used between chilli and PacketFence
+    HS_UAMHOMEPAGE  Comment that line from the config file and the defaults file to speedup login page
+    HS_UAMFORMAT    http://PACKETFENCE_MANAGEMENT_IP
+    HS_SSID         SSID name
+    HS_NASIP        Access-point IP address
+    HS_COAPORT      3799
+
+    Take note of the HS_UAMLISTEN value, since this will be required in the PacketFence configuration step.
+
+A startup script maybe required depending on the hardware. Again, a quick search on Google might be the best solution to find the best one
```
Comment (extrafu) [inline docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc]: may be

Comment (extrafu) [inline docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc]: hardware -> equipment vendor (everywhere)

Comment (extrafu) [inline docs/PacketFence_CoovaChilli_Quick_Integration_Guide.asciidoc]: a quick documentation search on the Internet might be the ...

File: lib/pf/web/constants.pm [hunk 16]
```diff
@@ -100,6 +100,7 @@ Readonly::Scalar our $REQ_ARUBA                 => 'apname';
 Readonly::Scalar our $REQ_CISCO_PORTAL          => '/cep(.*)';
 Readonly::Scalar our $REQ_RUCKUS                => 'sip';
 Readonly::Scalar our $REQ_AEROHIVE              => 'RADIUS-NAS-IP';
+Readonly::Scalar our $REQ_COOVACHILLI           => 'uamip';
```
Comment (julsemaan) [inline lib/pf/web/constants.pm]: As discussed verbally, just change this to use the EXT_URL method and we're good to merge

**Comment by dwlfrth** [comment]
Done

**Pull request merged**